File: LostAndFound-API/Domain/Models/Item.cs
using LostAndFound_API.Domain.Models.Identity;
using System.ComponentModel;

namespace LostAndFound_API.Domain.Models
{
    public class Item
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Color { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty; 
        public string LostLocationAddress { get; set; } = string.Empty;
        public string RetrievalLocationAddress { get; set; } = string.Empty;
        public string? Note { get; set; }
        public string? PictureUrl { get; set; }
        public string? PublicId { get; set; }
        public bool IsActive { get; set; }
        public DateTime CreatedAt { get; set; }
        public ModeOfContact  ModeOfContact { get; set; }
        public long ReportedByUserId { get; set; }
        public virtual User ReportedByUser { get; set; }
        public long? ClaimedByUserId { get; set; }
        public virtual User? ClaimedByUser { get; set; }
        public string? ClosingRemarks { get; set; }

        public virtual ICollection<ItemComment> ItemComments { get; set; }
        public virtual ICollection<UserItemBookmark> UserItemBookmarks { get; set; }

        public Item()
        {
            ItemComments = new HashSet<ItemComment>();
            UserItemBookmarks = new HashSet<UserItemBookmark>();
        }

    }

    public enum ModeOfContact 
    {
        [Description("Email")]
        Email = 1,

        [Description("Phone")]
        Phone =2,
    }
}

File: lostandfound/app/src/main/java/com/example/lostandfound/models/Item.java
package com.example.lostandfound.models;

public class Item {

    private int id, reportedByUserId;
    private String name, color, description, lostLocationAddress, retrievalLocationAddress, note, pictureUrl, contact, createdAt;
    private boolean active;

    public Item(int id, String name, String color, String description, String lostLocationAddress, String retrievalLocationAddress, String note, String pictureUrl, String contact, String createdAt, boolean active, int reportedByUserId) {
        this.id = id;
        this.name = name;
        this.color = color;
        this.description = description;
        this.lostLocationAddress = lostLocationAddress;
        this.retrievalLocationAddress = retrievalLocationAddress;
        this.note = note;
        this.pictureUrl = pictureUrl;
        this.contact = contact;
        this.createdAt = createdAt;
        this.active = active;
        this.reportedByUserId = reportedByUserId;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getColor() {
        return color;
    }

    public String getDescription() {
        return description;
    }

    public String getLostLocationAddress() {
        return lostLocationAddress;
    }

    public String getRetrievalLocationAddress() {
        return retrievalLocationAddress;
    }

    public String getNote() {
        return note;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public String getContact() {
        return contact;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public boolean isActive() {
        return active;
    }

    public int getReportedByUserId() {
        return reportedByUserId;
    }
}
